src/SplitProvider.tsx

```diff
@@ -42,7 +43,10 @@ const SplitProvider = ({
   children,
   onImpression,
 }: ISplitProviderProps) => {
-  const [client, setClient] = useState<SplitIO.IClient | null>(null);
+  const [{ factory, client }, set] = useState({
```

Comment:
Let's call it `setProvider` instead of just `set`.

Reply:
Yeah couldn't think of a good name. Works for me.